network/lwip_raw_socket/lwip_socket.c

```diff
@@ -808,27 +808,70 @@ static int32_t lwip_socket_recvfrom(void *net, uint32_t sock_id, void *data,
 }
 
 /**
- * @brief Not implemented.
- * @param net - Not used.
- * @param sock_id - Not used.
- * @param addr - Not used.
+ * @brief Called once a connect() completes.
+ * @param net - lwip sockets layer specific descriptor.
+ * @param pcb - unused.
+ * @param err - connection result.
+ * @return -ENOSYS
+ */
+static err_t lwip_connect_callback(void *arg, struct tcp_pcb *pcb, err_t err)
+{
+	struct lwip_socket_desc *sock = arg;
```

Comment:
seeing how these APIs are static, I am not sure if we should check for null pointers...I suppose the upper layer is taking care of it?

Reply:
The upper layer is not checking `arg`. I've added the check in the `lwip_connect_function`, however this pointer is shared between multiple callbacks (`lwip_accept_callback()`, `lwip_recv_callback()`) and an invalid value will result in a fatal error (meaning the lwip stack won't be able to recover from this and any action such as connect, accept, packet RX, etc will be broken from that point). So, it's not much different that a null pointer access.